packages/aws-cdk-lib/aws-ec2/lib/vpc.ts

```diff
@@ -1630,7 +1771,48 @@ export class Vpc extends VpcBase {
       throw new Error('Incomplete Subnet Allocation; response array dose not equal input array');
     }
 
-    this.createSubnetResources(requestedSubnets, allocatedSubnets);
+    if (this.useIpv6) {
+      if (this.ipv6SelectedCidr === undefined) {
+        throw new Error('No IPv6 CIDR block associated with this VPC could be found');
+      }
+      if (this.ipv6IpAddresses === undefined) {
+        throw new Error('No IPv6 IpAddresses were found');
+      }
+      // create the IPv6 CIDR block
+      const subnetIpv6Cidrs = Fn.cidr(this.ipv6SelectedCidr, allocatedSubnets.length, (128 - 64).toString());
```

Comment:
This decision should probably be in `Ipv6Addresses`.

Reply:
For standard networks, the interface ID is required to be 64 bits long (RFC 4291), so the subnet will be 64 bits ... but yes, maybe move into the address/subnet provider, to at least be consistent with IPv4